lib/Sema/TypeCheckAttr.cpp

```diff
@@ -4066,9 +4066,24 @@ static bool checkFunctionSignature(
   // Check that generic signatures match.
   auto requiredGenSig = required.getOptGenericSignature();
   auto candidateGenSig = candidateFnTy.getOptGenericSignature();
-  if (!candidateGenSig.requirementsNotSatisfiedBy(requiredGenSig).empty()) {
-    return false;
+  // Check that the candidate signature's generic parameters are a subset of
+  // those of the required signature.
+  if (requiredGenSig && candidateGenSig) {
+    auto requiredGenParams = requiredGenSig.getGenericParams();
+    auto candidateGenParams = candidateGenSig.getGenericParams();
+    if (candidateGenParams.size() > requiredGenParams.size())
+      return false;
+    for (auto candidateGenParam : candidateGenSig.getGenericParams()) {
+      auto found = std::find(requiredGenParams.begin(),
+                             requiredGenParams.end(),
+                             candidateGenParam);
+      if (!found)
```

Comment:
find() returns an iterator, so you have to compare the result against requiredGenParams.end() and not nullptr. However, maybe just comparing the length of the two vectors is sufficient? We don't ever 'skip' generic parameters.

Reply:
Oh, I forgot.  Changed to comparing the length.